src/engine/renderer/Material.cpp

```diff
@@ -705,6 +705,13 @@ void MaterialSystem::GenerateWorldMaterialsBuffer() {
 			const uint32_t padding = ( offset % paddedSize == 0 ) ? 0 : paddedSize - ( offset % paddedSize );
 
 			offset += padding;
+			
+			// Make sure padding is taken into account for dynamicDrawSurfsOffset
+			if ( !dynamicDrawSurfOffsetSet ) {
+				dynamicDrawSurfsOffset = offset;
```

Comment:
So this variable needs to point to the end of the first material? That seems weird.

Reply:
To the start of the first dynamic material, rather. Since the buffer is structured like this:
```
staticMaterial_0
...
staticMaterial_last
padding
dynamicMaterial_0
...
```
Padding after material `n` depends on the material `n + 1`, and is added at the `offset += padding;` line (last dynamic material never has any padding after it). `dynamicDrawSurfsOffset` is then used as the start of the range when mapping the buffer to update it, so it has to point to the index directly after the last static material padding.